backtrace-library/src/main/java/backtraceio/library/services/BacktraceNativeCrashResponseListener.java

```diff
@@ -0,0 +1,128 @@
+                for (String attachment : attachments) {
+                    if (attachment.endsWith(".dmp")) {
+                        moveToCompleted(attachment);
+                        break;
+                    }
+                }
+            }
+        }
+    }
+
+    public static void UploadMinidumps(BacktraceBase client, String databasePath) {
+        if (client == null) {
+            BacktraceLogger.e(LOG_TAG, "Backtrace client can't be null");
+            return;
+        }
+        if (databasePath.equals("")) {
+            BacktraceLogger.e(LOG_TAG, "Database path cannot be null");
+            return;
+        }
+        if (instance == null) {
+            instance = new BacktraceNativeCrashResponseListener();
+        }
+        Thread sender = new Thread(() -> {
+            List<File> minidumps = scanForNewMinidumps(databasePath);
+            if (minidumps.size() > 0) {
+                for (File minidump: minidumps) {
+                    BacktraceReport report = createBacktraceReport(minidump, databasePath);
+                    client.send(report, instance);
+                }
+            }
+        });
+        sender.run();
+    }
+
+    private static List<File> scanForNewMinidumps(String databasePath) {
+        List<File> minidumps = new ArrayList<>();
+        FilenameFilter minidumpFilter = (dir, name) -> name.endsWith(".dmp");
+        String[] newFiles = new File(databasePath + "/" + BacktraceConstants.NewFolder).list(minidumpFilter);
+        String[] pendFiles = new File(databasePath + "/" + BacktraceConstants.PendingFolder).list(minidumpFilter);
+        if (newFiles != null) {
+            for (String file : newFiles) {
+                minidumps.add(new File(databasePath + "/" + BacktraceConstants.NewFolder + "/" + file));
+            }
+        }
+        if (pendFiles != null) {
+            for (String file : pendFiles) {
+                minidumps.add(new File(databasePath + "/" + BacktraceConstants.PendingFolder + "/" + file));
+            }
+        }
+        return minidumps;
+    }
+
+    private static BacktraceReport createBacktraceReport(File minidump, String databasePath) {
+        BacktraceReport report = new BacktraceReport("");
+        report.attachmentPaths.add(minidump.getAbsolutePath());
+        String attachmentPath = databasePath + "/" + BacktraceConstants.AttachmentsFolder + "/"
+                + minidump.getName().replace(BacktraceConstants.MinidumpExtension, "");
+        File attachmentsDir = new File(attachmentPath);
+        if (attachmentsDir.isDirectory()) {
+            for (String file: attachmentsDir.list()) {
+                report.attachmentPaths.add(attachmentsDir.getAbsolutePath() + "/" + file);
+            }
+        }
+        return report;
+    }
+
+    private void moveToCompleted(String minidumpPath) {
+        String regex = "/(" + BacktraceConstants.NewFolder + "|" + BacktraceConstants.PendingFolder
```

Comment:
all of the variables in all of the methods in this file and also all of the params of methods can be final